liblwgeom/g_serialized.c

```diff
@@ -109,7 +110,7 @@ int32_t gserialized_get_srid(const GSERIALIZED *s)
 	if ( srid == 0 )
 		return SRID_UNKNOWN;
 	else
-		return clamp_srid(srid);
+		return srid;
```

Comment:
This is unrelated to "clean up sorting code path", isn't it ?
Could you please rebase and make sure only related changes are seen in the "Files changed" tab on github ?

Reply:
gserialized_get_srid is one of topmost not-inlined functions in sort:
![image](https://user-images.githubusercontent.com/810638/31273726-887be486-aa98-11e7-9665-d8b315f67bc2.png)

and it can't be inlined since it calls clamp_srid:
![image](https://user-images.githubusercontent.com/810638/31273767-c11e6228-aa98-11e7-8a3c-4899860f29dc.png)

Reply:
This optimization was really walked around later, as I made srid cmp function that doesn't construct SRID. But it also doesn't clamp it, which makes `cmp_srid(a,b)` and `get_srid(a)==get_srid(b)` diverge in opinions in edge cases. I think if we can't commit to non-clamping get_srid, we can't commit to non-clamping cmp_srid, so these both should go in pair.

Since we are clamping on write, without data corruption you shouldn't get non-clamped values on disk.

Reply:
I think the clamp-on-read was done for the restore case, when restoring a dump from pre-clamp dates (postgis_restore should take care of that, but pg_upgrade would not). Anyway this PR is about "clean up", not about "performance improvement", am I wrong ?

Reply:
Sorry for misleading wording. It is about performance, and addresses https://trac.osgeo.org/postgis/ticket/3864.

According to @robe2 pre-2.0 dumps come in HEXWKB, and for those clamping is active: https://github.com/postgis/postgis/blob/a5c7920a059c7273864760539b6845d74b34289c/liblwgeom/lwin_wkb.c#L701